src/lib/Microsoft.Health.Events/EventConsumers/Service/EventConsumerService.cs

```diff
@@ -35,44 +35,16 @@ public async Task ConsumeEvents(IEnumerable<IEventMessage> events)
             {
                 foreach (IEventConsumer eventConsumer in _eventConsumers)
                 {
-                    await OperationWithRetryAsync(eventConsumer, events);
-                }
-            }
-        }
-
-        private async Task OperationWithRetryAsync(IEventConsumer eventConsumer, IEnumerable<IEventMessage> events)
```

Comment:
>OperationWithRetryAsync [](http://example.com/codeflow?start=27&length=23)

Can we add this back in and make it optional in ctor?  Right now this will be break the FHIR Conversions.  Retryable errors like 429's or unable to auth to the FHIR service will error and cause loss of data.

Reply:
This would cause the FHIR Converter to retry forever on all exceptions. I'm assuming that is desired, since the Fhir Converter has custom exception handling logic? If thats the case I'll just move this into the Fhir Converter Processor as per @namalu comment moving it into the Startup.cs will be tricky given his refactor

Reply:
The FHIR converter doesn't retry forever on all exceptions.  It uses the [https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs](https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs) so known exceptions are "handled" and converted to telemetry that is emitted.  Basically any exception that makes it this far will cause it to retry indefinititely but that is intended.  Non-recoverable errors due to bad data are supposed to be caught and handled in the ExceptionTelemetryProcessor.

Reply:
Yup; sorry that is what I had meant; it'll retry forever on all exceptions not handled by that custom exception logic.